lnwallet/channel.go

```diff
@@ -4156,24 +4156,36 @@ func (lc *LightningChannel) ReceiveNewCommitment(commitSig lnwire.Sig,
 	return nil
 }
 
-// FullySynced returns a boolean value reflecting if both commitment chains
-// (remote+local) are fully in sync. Both commitment chains are fully in sync
-// if the tip of each chain includes the latest committed changes from both
-// sides.
-func (lc *LightningChannel) FullySynced() bool {
+// OweCommitment returns a boolean value reflecting whether we need to send
+// out a commitment signature because there are outstanding local updates and/or
+// updates in the local commit tx that aren't reflected in the remote commit tx
+// yet.
+func (lc *LightningChannel) OweCommitment() bool {
 	lc.RLock()
 	defer lc.RUnlock()
 
-	lastLocalCommit := lc.localCommitChain.tip()
 	lastRemoteCommit := lc.remoteCommitChain.tip()
+	lastLocalCommit := lc.localCommitChain.tip()
+
+	// There are local updates pending if our local update log is not in
+	// sync with our remote commitment tx.
+	localUpdatesPending := lc.localUpdateLog.logIndex !=
+		lastRemoteCommit.ourMessageIndex
 
-	localUpdatesSynced := (lastLocalCommit.ourMessageIndex ==
-		lastRemoteCommit.ourMessageIndex)
+	// There are remote updates pending if their remote commitment tx (our
+	// local commitment tx) contains updates that we don't have added to our
+	// remote commitment tx yet.
+	remoteUpdatesPending := lastLocalCommit.theirMessageIndex !=
+		lastRemoteCommit.theirMessageIndex
+
+	// If any of the conditions above is true, we owe a commitment
+	// signature.
+	oweCommitment := localUpdatesPending || remoteUpdatesPending
```

Comment:
i see now, the local and remote PoV are reversed. is it possible to consolidate this with the empty commit sig check via a helper method, e.g. `oweCommitment(local bool) bool`?

Reply:
consolidate in accordance with your suggestion

Reply:
looks great, thank you!